src/Common/src/System/Net/Mail/MailAddressParser.cs

```diff
@@ -283,7 +302,7 @@ private static string ParseDisplayName(string data, ref int index, bool expectMu
 
                 // Do not include the Comma (if any), and because there were no bounding quotes, 
                 // trim extra whitespace.
-                displayName = data.SubstringTrim(index + 1, startingIndex - index);
+                displayName = data.Substring(index + 1, startingIndex - index).Trim();
```

Comment:
Was there a reason this changed? `SubstringTrim` will only allocate one string instead of two strings for the separate `Substring` and `Trim` operations.